include/lp_hash_table.h

```diff
@@ -0,0 +1,100 @@
+#ifndef DIL_LP_HASH_TABLE_H
+#define DIL_LP_HASH_TABLE_H
+
+#include <stdlib.h>
+#include <stdint.h>
+#include <assert.h>
+
+template <typename T>
```

Comment:
Constrain `T` to be only of type pointer. This is C++ wizardry and I'm not the biggest fan of it. But in this case it's useful.